frontend/src/metabase/querying/notebook/components/NativeQueryPreview/NativeQueryPreview.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useRef, useState } from "react";
+import { useUnmount } from "react-use";
+import { t } from "ttag";
+
+import { useGetNativeDatasetQuery } from "metabase/api";
+import { getErrorMessage } from "metabase/api/utils";
+import { CodeEditor } from "metabase/common/components/CodeEditor";
+import ExternalLink from "metabase/common/components/ExternalLink";
+import { formatNativeQuery } from "metabase/lib/engine";
+import { useSelector } from "metabase/lib/redux";
+import { language } from "metabase/query_builder/components/NativeQueryEditor/CodeMirrorEditor/language";
+import { getLearnUrl } from "metabase/selectors/settings";
+import { getShowMetabaseLinks } from "metabase/selectors/whitelabel";
+import {
+  ActionIcon,
+  Box,
+  Flex,
+  Icon,
+  Loader,
+  Stack,
+  Tooltip,
+} from "metabase/ui";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
+import type { UiParameter } from "metabase-lib/v1/parameters/types";
+
+import S from "./NativeQueryPreview.module.css";
+
+export function NativeQueryPreview({
+  question,
```

Comment:
you pass the `Lib.Query` here. It's better not to have a dependency on `Question` if possible.